File: backend/bbm384oyd/src/main/java/com/example/bbm384oyd/model/Semester.java
package com.example.bbm384oyd.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "semesters")
public class Semester {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String startDate;
    private String endDate;

    public Semester() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    @Override
    public String toString() {
        return "Semester{" +
                "id=" + id +
                ", startDate='" + startDate + '\'' +
                ", endDate='" + endDate + '\'' +
                '}';
    }
}

File: mobil/app/src/main/java/com/example/oyd/Models/Semester.kt
package com.example.oyd.Models

import com.google.gson.annotations.SerializedName

class Semester (
    @SerializedName("startDate")
    var startDate: String?,

    @SerializedName("endDate")
    var endDate: String?
    ){
    override fun toString(): String {
        return "Semester{" +
                "startDate='" + startDate + '\'' +
                ", endDate='" + endDate + '\'' +
                "}"
    }
}
